packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -1045,24 +1044,25 @@ class TextureAndroidViewController extends AndroidViewController {
   int? get textureId => _textureId;
 
   /// The size used to create the platform view.
-  Size? _initialSize;
+  final Completer<Size> _initialSize = Completer<Size>();
 
   /// The current offset of the platform view.
   Offset _off = Offset.zero;
 
+  @override
+  void setInitialSize(Size size) {
+    assert(_state == _AndroidViewState.waitingForSize, 'Android view is already sized. View id: $viewId');
+    _initialSize.complete(size);
```

Comment:
Can you explain why this has to be a completer? 
Can we just have `_initialSize = size`?

Reply:
before this change, `setSize()` also called `create()`, so it didn't need a `Completer`.  I refactored to make it a little easier to follow, so `setSize()` no longer calls `create()`.

Reply:
Maybe I'm missing something, why does `setInitialSize` need a completer to set _initialSize.
Can we just do
```dart
_initialSize = size;
```

Reply:
what happens if the size is given by the framework after `create()` was called?